### DISCUSSION TITLE
Check whether the current route is in a route group

### DISCUSSION THREAD

**Discussion by nnennajohn** [vercel/next.js#51527] [App Router]
Is there a way to check whether a route is in a route group, similar to expo-router's useSegments?

With expo-router's useSegment, if a route is in /(auth)/login, I am able to check like below:

```
const rootSegment = useSegments()[0];

if (
      // If the user is not signed in and the initial segment is not anything in the auth group.
      !user && rootSegment !== "(auth)"
    ) {
    // Do something
 }
```
How do I accomplish something similar in next app-router, by checking the route group, and not iterating over every nested route in that group?

Thank you.

[Expo-router useSegments](https://expo.github.io/router/docs/features/linking/#usesegments)

**Comment by pierremouchan** [comment]
Would love to have it as well!
